op-node/rollup/derive/engine_queue.go

```diff
@@ -212,20 +222,13 @@ func (eq *EngineQueue) Step(ctx context.Context) error {
 		return err
 	}
 	eq.origin = newOrigin
-	// make sure we track the last L2 safe head for every new L1 block
-	if err := eq.postProcessSafeL2(); err != nil {
-		return err
-	}
-	// try to finalize the L2 blocks we have synced so far (no-op if L1 finality is behind)
-	if err := eq.finalizer.OnDerivationL1End(ctx, eq.origin); err != nil {
-		return fmt.Errorf("finalizer OnDerivationL1End error: %w", err)
-	}
+
 	if next, err := eq.prev.NextAttributes(ctx, eq.ec.PendingSafeL2Head()); err == io.EOF {
 		return io.EOF
 	} else if err != nil {
 		return err
 	} else {
-		eq.safeAttributes = next
+		eq.attributesHandler.SetAttributes(next)
```

Comment:
I don't see a great way around this push/set right now, but it does feel a little awkward.